src/DynamoCoreWpf/ViewModels/PackageManager/PackagePathViewModel.cs

```diff
@@ -226,10 +226,15 @@ private void RemovePathAt(int index)
 
         private void CommitChanges(object param)
         {
-            setting.CustomPackageFolders = CommitRootLocations();
-            if (this.packageLoader != null)
+            var newpaths = CommitRootLocations();
+            //if paths are modified, reload packages and update prefs.
+            if (!setting.CustomPackageFolders.SequenceEqual(newpaths))
             {
-                this.packageLoader.LoadCustomNodesAndPackages(loadPackageParams, customNodeManager);
+                setting.CustomPackageFolders = newpaths;
+                if (this.packageLoader != null)
+                {
+                    this.packageLoader.LoadCustomNodesAndPackages(loadPackageParams, customNodeManager);
```

Comment:
And where was thic call taking place earlier that caused the packages to reload each time?